.github/workflows/checkmycode.yml

```diff
@@ -1,30 +1,165 @@
-# This workflow will do a clean install of node dependencies, cache/restore them, build the source code and run tests across different versions of node
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
-name: Node.js CI
-
+name: Testworkflow_CI
 on:
   push:
-    branches: [ main ]
+    branches:
+      - dev
+      - feature    
   pull_request:
-    branches: [ main ]
-
+  release:
+    types: [published, unpublished, created, edited, deleted, prereleased]     
 jobs:
-  build:
-
-    runs-on: ubuntu-latest
-
+  checkout_and_install:
     strategy:
       matrix:
-        node-version: [16.x]
-        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
-
+        node: ["12.22.1", "14.16.1", "15.14.0", "16.1.0"]        
```

Comment:
How did you select the above versions?

Reply:
Matrix node version helps to do the parallel execution on different npm versions, build task get invoked on all the listed npm supported version at same time, if we have any version dependencies related concern we can directly specify the versions as a value i.e. node: [16.1.0]